Modules/readline.c

```diff
@@ -126,6 +126,16 @@ readline_clear(PyObject *m)
    return 0;
 }
 
+static void
+readline_cleanup(void)
+{
+    rl_startup_hook = NULL;
+#ifdef HAVE_RL_PRE_INPUT_HOOK
+    rl_pre_input_hook = NULL;
+#endif
+    rl_attempted_completion_function = NULL;
```

Comment:
It seems like there are many more hooks and config in `setup_readline()` to clean up here.

Reply:
This function is removed now